numpy/core/code_generators/generate_umath.py

```diff
@@ -1291,23 +1301,42 @@ def make_ufuncs(funcdict):
             sig = "NULL"
         else:
             sig = '"{}"'.format(uf.signature)
-        fmt = textwrap.dedent("""\
-            identity = {identity_expr};
-            if ({has_identity} && identity == NULL) {{
-                return -1;
-            }}
-            f = PyUFunc_FromFuncAndDataAndSignatureAndIdentity(
-                {name}_functions, {name}_data, {name}_signatures, {nloops},
-                {nin}, {nout}, {identity}, "{name}",
-                {doc}, 0, {sig}, identity
-            );
-            if ({has_identity}) {{
-                Py_DECREF(identity);
-            }}
-            if (f == NULL) {{
-                return -1;
-            }}
-        """)
+        if name in empty:
+            fmt = textwrap.dedent("""\
+                identity = {identity_expr};
+                if ({has_identity} && identity == NULL) {{
+                    return -1;
+                }}
+                f = PyUFunc_FromFuncAndDataAndSignatureAndIdentity(
+                    NULL, NULL, NULL, {nloops},
```

Comment:
Only this line is different between the two stanzas. It should be possible to refactor this
- to select on an attribute of `uf` rather than have a separate `empty` list, passing it around seems awkward. Something like what I suggested above?
- extra points for tweaking `fmt` for an empty `uf` rather than rewriting the whole thing